### PR TITLE
feat(): selecting terminal fonts and shell

### PR DESCRIPTION
**Pull request by marhi**
This is the setup for the FontService, which will return a list of all suitable terminal font-families. It should work on MacOS, Linux and Windows.
This will enable the end-user to quickly and effortlessly select the terminal font.

### CODE REVIEW

**Comment by omerson** [comment]
Just a note, this should be tried in production mode. All new external dependencies need to be added in config.json dependencies section, otherwise they are unavailable in production mode.

**Comment by marhi** [comment]
fontselect

**Comment by jkuri** [comment]
works good 👍  I would just change the design before getting this in...

**Comment by jkuri** [comment]
@marhi can you rebase this one so this gets merged, then I can take care of UI, I think

File: e2e/app.e2e.ts [hunk 1]
```diff
@@ -557,4 +557,34 @@ describe('bterm launch', function() {
       .then(text => expect(text).to.contain(testString));
   });
 
+it('should copy and paste the text', () => {
```
Comment (marhi) [inline e2e/app.e2e.ts]: made by @omerson

File: src/app/services/system.service.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    this.fonts = [];
+    this.shells = [];
+  }
+
+  getFonts() {
+    if (this.fonts.length) { return Promise.resolve(this.fonts); }
+
+    return new Promise<IFonts[]>(resolve => {
+      fontManager.findFonts({ monospace: true }, result => {
+        result = result.map(r => ({ family: r.family, toString: function() { return this.family }}));
```
Comment (jkuri) [inline src/app/services/system.service.ts]: `() => this.family`

File: src/app/services/system.service.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      if (!this._config.config.settings.shells) { resolve ([]); }
+
+      this._config.config.settings.shells.forEach((s: string) => {
+        let foundShell: any = which(s);
+        if (foundShell) {
+          this.shells.push({
+            shell: foundShell.toString(),
+            args: [],
+            short: path.basename(foundShell.toString()),
+            toString: function() { return this.short }
```
Comment (jkuri) [inline src/app/services/system.service.ts]: `() => this.sort;`

Comment (marhi) [inline src/app/services/system.service.ts]: Actually, this.short will reference SystemService.short if it's used as an arrow function, while the usage of function will reference the correct parameter "short" inside the object.

**Comment by marhi** [comment]
The design needs some work, but I think the functionality is there.

**Pull request merged**